pkg/app/api/grpcapi/piped_api.go

```diff
@@ -679,6 +681,94 @@ func (a *PipedAPI) ReportApplicationLiveStateEvents(ctx context.Context, req *pi
 	return &pipedservice.ReportApplicationLiveStateEventsResponse{}, nil
 }
 
+// GetLatestEvent gives back the latest event that meets the given conditions.
+func (a *PipedAPI) GetLatestEvent(ctx context.Context, req *pipedservice.GetLatestEventRequest) (*pipedservice.GetLatestEventResponse, error) {
+	projectID, _, _, err := rpcauth.ExtractPipedToken(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// Try to fetch the most recently registered event.
+	opts := datastore.ListOptions{
+		PageSize: 1,
+		Filters: []datastore.ListFilter{
+			{
+				Field:    "ProjectId",
+				Operator: "==",
+				Value:    projectID,
+			},
+			{
+				Field:    "Name",
+				Operator: "==",
+				Value:    req.Name,
+			},
+			// TODO: Enable to use labels to filter Events
+		},
+		Orders: []datastore.Order{
+			{
+				Field:     "UpdatedAt",
+				Direction: datastore.Desc,
+			},
+		},
+	}
+	events, err := a.eventStore.ListEvents(ctx, opts)
+	if err != nil {
+		a.logger.Error("failed to list events", zap.Error(err))
+		return nil, status.Error(codes.Internal, "failed to list event")
+	}
+	if len(events) == 0 {
+		a.logger.Error("no events found",
```

Comment:
I think this shouldn't be an Error log.

Reply:
Right, I started thinking the log itself is no longer needed.